### PR TITLE
Verify xcframework integrations

### PR DESCRIPTION
**Pull request by bielikb**
### 🎯 Goal
Verify xcframework integration for SPM, CocoaPods and Carthage

### 🛠 Implementation
New folder `XCIntegration` contains standalone `StreamChatIntegration-XC.xcworkspace`.
All new projects/targets/schemes carry `XC` suffix.

### 🎨 Changes

* Adds `StreamChatIntegration-XC.xcworkspace` with respective projects for Carthage, CocoaPods, SPM.
* Adds new lane `build_xc_integration_app` that builds xc integration apps.
* Adds new lane `build_all_integration_apps` that can be leveraged locally to build all the integration apps.
* All 3 new integration apps were added to the GHA matrix.
* Script `fetch_dependencies.sh` is now capable of fetching xc related dependencies.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -0,0 +1 @@
+github "GetStream/stream-chat-swift" "4.6.0"
```
Comment (polqf) [inline Cartfile.resolved:1]: Based on my experience, it is better not to commit this file. Seems like it makes dependency resolution more complex in some cases

File: XCIntegration/Cartfile.resolved [hunk 2]
```diff
@@ -0,0 +1 @@
+binary "https://raw.githubusercontent.com/GetStream/stream-chat-swift/main/StreamChatArtifacts.json" "4.6.0"
```
Comment (polqf) [inline XCIntegration/Cartfile.resolved:1]: same as before

File: scripts/fetch_dependencies.sh [hunk 3]
```diff
@@ -25,50 +25,127 @@ do
   esac
 done
 
+XC_FOLDER=XCIntegration
+
 function updatePods {
-  echo "> Grabbing SDKs using CocoaPods integration."
```
Comment (polqf) [inline scripts/fetch_dependencies.sh:29]: Why are you removing the echos from its own function? I don't see any benefit on having different logs for when we pass an integration parameter or when we don't

**Review by polqf — APPROVED** [review]
Would be nice to move the OSS files into its own folder as well

File: Cartfile.resolved (hunk 1 above)
Comment (bielikb) [reply Cartfile.resolved]: Done.

File: scripts/fetch_dependencies.sh (hunk 3 above)
Comment (bielikb) [reply scripts/fetch_dependencies.sh:29]: It's a safety measure. Depending on the version of bash used, echo can act as a `return` from a local func, e.g. when you want to return a value from a func you would just echo it.

**Review by polqf — APPROVED** [review]

**Pull request merged**